nixos/modules/installer/tools/nixos-rebuild.sh

```diff
@@ -4,6 +4,11 @@ if [ -x "@shell@" ]; then export SHELL="@shell@"; fi;
 
 set -e
 
+if [[ $EUID > 0 ]]; then
+  echo -e "\033[31;1merror:\033[0m Permission denied. Try running nixos-rebuild as root." >&2
+  exit 1
+fi
```

Comment:
`nixos-rebuild build` does not require root privileges. There are also other commands that do not require root privileges, so this must be applied only conditionally.